### DISCUSSION TITLE
Install discogs_client through pipx

### DISCUSSION THREAD

**Discussion by brunescio** [joalla/discogs_client#147] [General] [answered]
When I try to install the client through pipx I get the following message: "No apps associated with package python3-discogs-client."

The [pipx documentation](https://pipx.pypa.io/stable/how-pipx-works/#developing-for-pipx) might provide an explanation but I'm not knowledgeable enough to really understand what it means.

What's the best way to install discogs_client if I can't use pip?

Thank you!

(I installed Python with Homebrew and it seems that pipx is the recommended way to install Python packages)

**Comment by JOJ0** [comment]
Is this the exact error message you get: https://stackoverflow.com/questions/60032087/why-my-package-cant-be-installed-with-pipx

I'm not very familiar with pipx, and only can repeat what's said there: pipx seems to look for entry_points section in our setup.py. We dont have that because we are a python library and not an executable (commandline) tool.

Maybe I can help in another way: Why can't u use pip?

**Reply by brunescio** [reply]
Thank you for your answer.

Yes, it's the same error message.

My Python was installed with Homebrew. If I use pip I get the message shown below. 

I'm not familiar with virtual environments but maybe that's the best option.

```
error: externally-managed-environment

× This environment is externally managed
╰─> To install Python packages system-wide, try brew install
    xyz, where xyz is the package you are trying to
    install.
    
    If you wish to install a Python library that isn't in Homebrew,
    use a virtual environment:
    
    python3 -m venv path/to/venv
    source path/to/venv/bin/activate
    python3 -m pip install xyz
    
    If you wish to install a Python application that isn't in Homebrew,
    it may be easiest to use 'pipx install xyz', which will manage a
    virtual environment for you. You can install pipx with
    
    brew install pipx
    
    You may restore the old behavior of pip by passing
    the '--break-system-packages' flag to pip, or by adding
    'break-system-packages = true' to your pip.conf file. The latter
    will permanently disable this error.
    
    If you disable this error, we STRONGLY recommend that you additionally
    pass the '--user' flag to pip, or set 'user = true' in your pip.conf
    file. Failure to do this can result in a broken Homebrew installation.
    
    Read more about this behavior here: <https://peps.python.org/pep-0668/>

note: If you believe this is a mistake, please contact your Python installation or OS distribution provider. You can override this, at the risk of breaking your Python installation or OS, by passing --break-system-packages.
hint: See PEP 668 for the detailed specification.
```

**Reply by JOJ0** [reply]
If you plan to use p3dc in your coding project, using a venv is a typical workflow and worth learning.

**Reply by JOJ0** [reply]
I'll lokk into the external managed error snyway. thanks for sharing :-)

**Reply by JOJ0** [reply]
also like to suggest you

brew install pyenv

It's a nice helper for setting up venvs.

HTH 🙂

**Reply by brunescio** [reply, marked as answer]
I created a virtual environment and "pip installed" discogs_client there. Works like a charm.

Thank you for your support!
